### PR TITLE
Add new optional params for datasources uploading

### PR DESCRIPTION
**Pull request by blokhin**

### CODE REVIEW

File: metis_client/metis.py [hunk 1]
```diff
@@ -74,9 +74,9 @@ class MetisDatasourcesNamespaceSync(MetisNamespaceSyncBase):
     """Datasources endpoints namespace"""
 
     @to_sync_with_metis_client
-    async def create(self, client: MetisAPIAsync, content: str):
+    async def create(self, client: MetisAPIAsync, content: str, fmt: str = None, name: str = None):
```
Comment (knopki) [inline metis_client/metis.py:77]: `Optional[str] = None`

File: metis_client/namespaces/datasources.py [hunk 2]
```diff
@@ -19,20 +19,20 @@
 class MetisDatasourcesNamespace(BaseNamespace):
     """Datasources endpoints namespace"""
 
-    async def create_event(self, content: str) -> MetisRequestIdDTO:
+    async def create_event(self, content: str, fmt: str = None, name: str = None) -> MetisRequestIdDTO:
```
Comment (knopki) [inline metis_client/namespaces/datasources.py:22]: `Optional[str] = None`

File: metis_client/namespaces/datasources.py [hunk 3]
```diff
@@ -19,20 +19,20 @@
...
             url=self._base_url,
-            json={"content": content},
+            json={"content": content, "fmt": fmt, "name": name},
             auth_required=True,
         ) as resp:
             return await resp.json()
 
-    async def create(self, content: str) -> Optional[MetisDataSourceDTO]:
-        "Create data source and wait for result"
+    async def create(self, content: str, fmt: str = None, name: str = None) -> Optional[MetisDataSourceDTO]:
```
Comment (knopki) [inline metis_client/namespaces/datasources.py:32]: `Optional[str] = None`

File: metis_client/namespaces/datasources.py [hunk 4]
```diff
@@ -19,20 +19,20 @@
...
     """Datasources endpoints namespace"""
 
-    async def create_event(self, content: str) -> MetisRequestIdDTO:
+    async def create_event(self, content: str, fmt: str = None, name: str = None) -> MetisRequestIdDTO:
         "Create data source"
         async with self._client.request(
             method="POST",
             url=self._base_url,
-            json={"content": content},
+            json={"content": content, "fmt": fmt, "name": name},
```
Comment (knopki) [inline metis_client/namespaces/datasources.py:27]: So, server-side formatting?
Now only "content" field is used by server. Is it ok?

**Review by knopki — CHANGES_REQUESTED** [review]
What is the meaning of this new fields?
Also, formatting is needed. Maybe add parameters in tests.

File: metis_client/namespaces/datasources.py (hunk 4 above)
Comment (blokhin) [reply metis_client/namespaces/datasources.py:29]: explained below in the referring commits

File: metis_client/metis.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 import sys
 from functools import partial
-from typing import Any, TypeVar, cast
+from typing import Any, TypeVar, cast, Optional
```
Comment (knopki) [inline metis_client/metis.py:5]: `isort` thing
```suggestion
from typing import Any, Optional, TypeVar, cast
```

**Review by knopki — APPROVED** [review]
LGTM

**Pull request merged**